solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -1136,10 +1137,23 @@ private SolrCore(
       updateProcessorChains = loadUpdateProcessorChains();
       reqHandlers = new RequestHandlers(this);
       reqHandlers.initHandlersFromConfig(solrConfig);
-      jerseyAppHandler =
-          (V2ApiUtils.isEnabled())
-              ? new ApplicationHandler(reqHandlers.getRequestHandlers().getJerseyEndpoints())
-              : null;
+      if (V2ApiUtils.isEnabled()) {
+        final String effectiveSolrConfigId = solrConfig.effectiveId(configSet.getName());
+        jerseyAppHandler =
+            coreContainer
+                .getAppHandlerCache()
+                .computeIfAbsent(
+                    effectiveSolrConfigId,
+                    () -> {
+                      log.debug(
```

Comment:
add log.isDebugEnabled()`

Reply:
Why?  Note we have checks that can tell when it's helpful and when it isn't.  It isn't helpful in this circumstance because the interpolated arguments are plain references -- not method calls that might do expensive stuff.  The log.isDebugEnabled is verbose IMO for too little gain.